persist/fs/write.go

```diff
@@ -195,33 +235,141 @@ func (w *writer) writeAll(
 			return err
 		}
 	}
-	if _, err := w.indexFdWithDigest.WriteBytes(w.encoder.Bytes()); err != nil {
-		return err
-	}
+
+	w.indexEntries = append(w.indexEntries, entry)
 	w.currIdx++
 
 	return nil
 }
 
-func (w *writer) close() error {
+func (w *writer) writeIndexEntriesSummariesBloomFilter() error {
```

Comment:
Seems like this function does more than the name implies. Maybe: "writeIndexEntriesAndIndexSummaryAndBloomFilter" or something. I know that's verbose but the existing name makes it seem like its writing one file when its writing 3

Reply:
Yeah I might break some of this up into smaller fns and have a second attempt at naming this.

Reply:
+1, breaking into smaller fns would help readability.

Reply:
Done.